Zend/zend_compile.c

```diff
@@ -5297,6 +5297,24 @@ void zend_compile_match(znode *result, zend_ast *ast)
 	uint32_t cond_count = 0;
 	uint32_t *jmp_end_opnums = safe_emalloc(sizeof(uint32_t), arms->children, 0);
 
+	// The generated default arm is emitted first to avoid live range issues where the tmpvar
+	// for the arm result is freed even though it has not been initialized yet.
+	if (!has_default_arm) {
+		if (!uses_jumptable) {
+			zend_update_jump_target_to_next(opnum_default_jmp);
```

Comment:
Wait, what does this do when there isn't a jump table?  I thought this should only be first when there was a jump table, and the last in the IDENTICAL check chain when there is no jump table

Reply:
Never mind, I forgot how the relocation worked

Reply:
I had that exact thought at first actually 🙂 But since the jump chain and arm bodies are separate luckily we can handle this the same.